src/lib/fileutils.test.ts

```diff
@@ -236,3 +253,73 @@ describe("generating service Dockerfile", () => {
     );
   });
 });
+
+describe("starterAzurePipelines", () => {
+  // Create a random workspace dir before every test
+  let randomDirPath = "";
+  beforeEach(() => {
+    randomDirPath = path.join(os.tmpdir(), uuid());
```

Comment:
note: i've had to use manual fs creation as opposed to `mock-fs` because it doesn't play nicely with `path.resolve` which `generateStarterAzurePipelinesYaml` uses